tests/unit/artifact/test_file.py

```diff
@@ -0,0 +1,75 @@
+import tempfile
+from unittest.mock import MagicMock
+
+import pytest
+
+from tmt.log import Logger
+from tmt.steps.prepare.artifact.providers.file import (
+    PackageAsFileArtifactInfo,
+    PackageAsFileArtifactProvider,
+)
+from tmt.utils import Path
+
+
+@pytest.mark.parametrize(
+    ("pattern", "expected_count", "expected_names"),
+    [
+        # Single file
+        ("foo.rpm", 1, {"foo.rpm"}),
+        # Glob pattern matching 3 files
+        ("package-*.rpm", 3, {f"package-{i}.rpm" for i in range(3)}),
+    ],
+)
+def test_file_artifact_provider_patterns(root_logger, pattern, expected_count, expected_names):
+    with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
Instead of `tempfile.*` use [pytest's fixture](https://docs.pytest.org/en/latest/reference/fixtures.html#reference-fixtures) in this case `tmp_path`